yapapi/executor/events.py

```diff
@@ -278,3 +267,36 @@ class DownloadFinished(Event):
 @dataclass
 class ShutdownFinished(HasExcInfo):
     """Indicates the completion of Executor shutdown sequence"""
+
+
+@dataclass
+class CommandEventContext:
+    evt_cls: Type[CommandEvent]
+    kwargs: dict
+
+    def computation_finished(self, last_idx: int) -> bool:
+        return self.evt_cls is CommandExecuted and (
+            self.kwargs["cmd_idx"] >= last_idx or not self.kwargs["success"]
+        )
+
+    def event(self, agr_id: str, task_id: str, cmds: List) -> CommandEvent:
+        kwargs = dict(agr_id=agr_id, task_id=task_id, **self.kwargs)
+        if self.evt_cls is CommandExecuted:
+            kwargs["command"] = cmds[self.kwargs["cmd_idx"]]
+            kwargs["stdout"] = kwargs["stderr"] = None
+            try:
+                # Replace the "message" kwarg with "stdout" and "stderr"
```

Comment:
hmm.... is this by design that we're allowing arbitrary other keyword arguments to pass through this function?

Reply:
Well, I'm not the original author of this piece of code but I guess the reason is that `kwargs` are used to provide arguments for calls to `evt_cls.__init__()`, where `evt_cls` is some subclass of `CommandEvent`, known only in runtime. So it's not possible to specify statically which `kwargs` should be allowed.